api/lib/models/location/location-data-mapper-impl.ts

```diff
@@ -0,0 +1,170 @@
+  public readonly READ_ALL: string = 'location:readall';
+  public readonly COUNT: string = 'location:count';
+  public tableName = 'LOCATIONS';
+
+  protected pkColumnName = 'uid';
+
+  constructor(
+    @Inject('IAcl') acl: IAcl,
+    @Inject('MysqlUow') protected readonly sql: MysqlUow,
+    @Inject('BunyanLogger') protected readonly logger: Logger,
+  ) {
+    super(acl);
+    super.addRBAC(
+      [this.CREATE, this.UPDATE, this.DELETE],
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    );
+    super.addRBAC([this.READ, this.READ_ALL], [AuthLevel.PARTICIPANT]);
+  }
+
+  public delete(id: UidType): Promise<IDbResult<void>> {
+    const query = squel
+      .delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+      .where(`${this.pkColumnName} = ?`, id)
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    )
+      .pipe(map(() => ({ result: 'Success', data: undefined })))
+      .toPromise();
+  }
+
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<Location>> {
+    let queryBuilder = squel
+      .select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder.where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query<Location>(query.text, query.values, {
+        cache: true,
+        stream: false,
+      }),
+    )
+      .pipe(
+        map((location: Location) => ({ result: 'Success', data: location })),
+      )
+      .toPromise();
+  }
+
+  public getAll(): Promise<IDbResult<Stream<Location>>> {
+    const query = squel
+      .select({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName, 'location')
+      .field('location.*')
+      .field('location.location_name')
+      .join('HACKATHON', 'h', 'h.uid=location.hackathon and h.active=true')
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<Location>(query, [], { stream: true, cache: true }),
+    )
+      .pipe(
+        map((location: Stream<Location>) => ({
+          data: location,
+          result: 'Success',
+        })),
+      )
+      .toPromise();
+  }
+
+  public getCount(): Promise<IDbResult<number>> {
+    const query = squel
+      .select({ autoQuoteTableNames: true, autoQuoteFieldNames: false })
+      .from(this.tableName)
+      .field(`COUNT(${this.pkColumnName})`, 'count')
+      .toString()
+      .concat(';');
+    const params = [];
+    return from(
+      this.sql.query<number>(query, params, { stream: true, cache: true }),
+    )
+      .pipe(map((result: number) => ({ result: 'Success', data: result })))

```

Comment:
A quick warning: the `mysql.query()` return type was changed in the main branch, so this and the `.get()` query will not compile once this branch is merged. It might be useful to change the type for the parameter for this `map()` to `map((result: number[]) => {...})` and likewise `map((location: Location[]) => {...})` for the `get()`. Another note that since it is an array, you will have to return the first element of the array to get the code to compile. Let me know if that is confusing, I can elaborate further.